### PR TITLE
refactor: group and group flows resources

### PR DESCRIPTION
**Pull request by haritianaadriano**

### CODE REVIEW

File: src/main/java/school/hei/haapi/endpoint/rest/mapper/GroupMapper.java [hunk 1]
```diff
@@ -41,7 +41,7 @@ public school.hei.haapi.model.notEntity.CreateGroup toDomain(CreateGroup restGro
         .CreateGroup
         .builder()
         .group(group)
-        .studentsToAdd(restGroup.getStudentsToAdd())
+        .studentsToAdd(restGroup.getStudents())
         .build();
```
Comment (haritianaadriano) [inline src/main/java/school/hei/haapi/endpoint/rest/mapper/GroupMapper.java]: rename to students

File: src/main/java/school/hei/haapi/model/Group.java [hunk 2]
```diff
@@ -45,6 +43,27 @@ public class Group implements Serializable {
...
+        + ", name='"
+        + name
+        + '\''
+        + ", ref='"
+        + ref
+        + '\''
+        + ", creationDatetime="
+        + creationDatetime
+        + ", awardedCourse="
+        + awardedCourse.stream().map(AwardedCourse::getId)
```
Comment (haritianaadriano) [inline src/main/java/school/hei/haapi/model/Group.java]: we do not include related entities in tostring

File: src/main/java/school/hei/haapi/model/Group.java [hunk 3]
```diff
@@ -45,6 +43,27 @@ public class Group implements Serializable {
...
+        + ", groupFlows="
+        + groupFlows.stream().map(GroupFlow::getId)
```
Comment (haritianaadriano) [inline src/main/java/school/hei/haapi/model/Group.java]: same rq

File: src/main/java/school/hei/haapi/service/GroupFlowService.java [hunk 4]
```diff
@@ -25,26 +26,58 @@ public class GroupFlowService {
...
+                });
+    User student =
+        userRepository
+            .findById(createGroupFlow.getStudentId())
+            .orElseThrow(
+                () -> {
+                  throw new NotFoundException(
+                      "student with id." + createGroupFlow.getStudentId() + " not found");
+                });
+
```
Comment (haritianaadriano) [inline src/main/java/school/hei/haapi/service/GroupFlowService.java]: use services

File: src/main/java/school/hei/haapi/service/GroupFlowService.java [hunk 5]
```diff
@@ -25,26 +26,58 @@ public class GroupFlowService {
...
             .group(group)
             .flowDatetime(Instant.now())
             .student(student)
             .groupFlowType(
                 GroupFlow.GroupFlowType.fromValue(createGroupFlow.getMoveType().getValue()))
             .build();
-    validator.accept(groupFlow);
-    return repository.save(groupFlow);
+    validator.accept(toSave);
+
```
Comment (haritianaadriano) [inline src/main/java/school/hei/haapi/service/GroupFlowService.java:39]: rm

File: src/main/java/school/hei/haapi/service/GroupService.java [hunk 6]
```diff
@@ -52,8 +52,9 @@ public List<Group> saveAll(List<school.hei.haapi.model.notEntity.CreateGroup> cr
     List<school.hei.haapi.model.Group> groups = new ArrayList<>();
     List<CreateGroupFlow> createGroupFlows = new ArrayList<>();
     for (school.hei.haapi.model.notEntity.CreateGroup createGroup : createGroups) {
-      Group group = repository.save(createGroup.getGroup());
+      Group group = createGroup.getGroup();
       groups.add(group);
+
       if (createGroup.getStudentsToAdd() != null) {
         for (String studentId : createGroup.getStudentsToAdd()) {
           createGroupFlows.add(
```
Comment (haritianaadriano) [inline src/main/java/school/hei/haapi/service/GroupService.java]: usee map

File: src/main/java/school/hei/haapi/service/GroupFlowService.java [hunk 7]
```diff
@@ -20,41 +21,40 @@
...
-              .build());
-    }
-    validator.accept(groupFlows);
-    return repository.saveAll(groupFlows);
+    List<GroupFlow> groupFlowsToSave =
+        createGroupFlows.stream().map(this::mapToDomainGroupFlow).collect(Collectors.toList());
+    return repository.saveAll(groupFlowsToSave);
+  }
+
+  private GroupFlow mapToDomainGroupFlow(CreateGroupFlow toMap) {
```
Comment (Mahefaa) [inline src/main/java/school/hei/haapi/service/GroupFlowService.java]: rename method name to "from"

File: src/main/java/school/hei/haapi/service/GroupFlowService.java [hunk 8]
```diff
@@ -20,41 +21,40 @@
...
+        .findById(userId)
+        .orElseThrow(() -> new NotFoundException("User with id." + userId + " not found"));
+  }
+
+  private Group findGroupById(String groupId) {
+    return groupRepository
+        .findById(groupId)
+        .orElseThrow(() -> new NotFoundException("Group with id." + groupId + " not found"));
+  }
+
```
Comment (Mahefaa) [inline src/main/java/school/hei/haapi/service/GroupFlowService.java:39]: are these methods used ?

**Review by Mahefaa — CHANGES_REQUESTED** [review]

**Pull request merged**